### PR TITLE
[10.0][sale_exception] Refactore : Adapt sale_exception to base_exception simplification

### PR DESCRIPTION
**Pull request by florian-dacosta**
Depends on : https://github.com/OCA/server-tools/pull/1586
It should be merged right after the PR on base_exception is merged, else sale_exception would be broken.

Careful before merging, I have adapted the oca_dependencies with base_exception PR to make the test pass, it should be modified right before the merge...

@hparfr @sebastienbeau @mourad-ehm

### CODE REVIEW

**Comment by florian-dacosta** [comment]
Error does not seems link to this PR.

**Review by sebastienbeau — APPROVED** [review]
Code review, It's ok regarding the discution done here : https://github.com/OCA/server-tools/pull/1586

**Review by PierrickBrun — APPROVED** [review]

File: oca_dependencies.txt [hunk 1]
```diff
@@ -2,7 +2,7 @@ bank-payment
 stock-logistics-workflow
 stock-logistics-transport
 account-closing
-server-tools
+server-tools https://github.com/akretion/server-tools 10-refactore-base-exception
```
Comment (PierrickBrun) [inline oca_dependencies.txt]: revert this

**Review by bealdav — APPROVED** [review]

File: oca_dependencies.txt (hunk 1 above)
Comment (florian-dacosta) [reply oca_dependencies.txt]: Done

**Pull request merged**